README.md

```diff
@@ -188,6 +188,141 @@ For the deidentification of the EHR extracts, we rely on an instance running the
+(set up instructions below).
+PIXL will query this key vault for the required secrets at runtime. This requires the following
+environment variables to be set in `.secrets.env` so that PIXL can connect to the key vault:
+
+- `EXPORT_AZ_CLIENT_ID`: the service principal's client ID, mapped to `AZURE_CLIENT ID` in `docker-compose`
+- `EXPORT_AZ_CLIENT_PASSWORD`: the password, mapped to `AZURE_CLIENT_SECRET` in `docker-compose`
+- `EXPORT_AZ_TENANT_ID`: ID of the service principal's tenant. Also called its 'directory' ID. Mapped to `AZURE_TENANT_ID` in `docker-compose`
+- `EXPORT_AZ_KEY_VAULT_NAME` the name of the key vault, used to connect to the correct key vault
+
+<details><summary>Azure Keyvault setup</summary>
+
+## Azure Keyvault setup
+
+_This is done for the \_UCLH_DIF_ `dev` tenancy, will need to be done once in the _UCLHAZ_ `prod`
+tenancy when ready to deploy to production.\_
+
+This Key Vault and secret must persist any infrastructure changes so should be separate from disposable
+infrastructure services. ServicePrincipal is required to connect to the Key Vault.
+
+The application uses the ServicePrincipal and password to authenticates with Azure via environment
+variables. See [here](https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.environmentcredential?view=azure-python)
+for more info.
+
+The Key Vault and ServicePrincipal have already been created for the `dev` environment and details
+are stored in the `pixl-secrets` note in the shared PIXL folder on _LastPass_.
+
+The process for doing so using the `az` CLI tool is described below.
+This process must be repeated for `staging` & `prod` environments.
+
+### Step 1
+
+Create the Azure Key Vault in an appropriate resource group:
+
+```bash
+az keyvault create --resource-group <resource-group-name> --name <key-vault-name> --location "UKSouth"
+```
+
+### Step 2
+
+Create Service Principal & grant access as per
+
+```bash
+az ad sp create-for-rbac -n hasher-api --skip-assignment
+```
+
+This will produce the following output
+
+```json
+{
+    "appId": "<generated-app-ID>",
+    "displayName": "<app-name>",
+    "name": "http://<app-name>",
+    "password": "<generated-password>",
+    "tenant": "<tenant-ID>"
+}
+```
+
+### Step 3
+
+Assign correct permissions to the newly created ServicePrincipal
+
+```bash
+az keyvault set-policy --name <key-vault-name> --spn <generated-app-ID> --secret-permissions backup delete get list set
+```
+
+### Step 4
+
+Create a secret and store in the Key Vault
+
+Use Python to create a secret:
+
+```python
+import secrets
+secrets.token_urlsafe(32)
+```
+
+copy the secret and paste as <secret-value> below
+
+```bash
+az keyvault secret set --vault-name "<key-vault-name>" --name "<secret-name>" --value "<secret-value>"
+```
+
+### Step 5
+
+Save credentials in `.secrets.env` and a LastPass `PIXL Keyvault <environment> secrets` note.
```

Comment:
have we done this step?

Reply:
You mean the LastPass? Guess it wouldn't be a bad idea to track projects? But maybe change "`<environment>`" to `<project>`